eval/eval.go

```diff
@@ -785,6 +785,9 @@ func evalStringInfixExpression(operator token.Type, left, right object.Object) o
 	case operator == token.ASTERISK && right.Type() == object.INTEGER:
 		rightVal := right.(object.Integer).Value
 		n := len(leftVal) * int(rightVal)
+		if rightVal < 0 {
+			return s.NewError("right operand of * on strings must be a positive integer")
```

Comment:
Is there a need for the zero?

Reply:
yes *0 is useful/valid and gets empty string (or empty array or empty map)